pkgs/applications/blockchains/bitbox/default.nix

```diff
@@ -0,0 +1,73 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, buildNpmPackage
+, clang
+, go
+, qtwebengine
+, wrapQtAppsHook
+}:
+
+let
+  src = fetchFromGitHub {
+    owner = "digitalbitbox";
+    repo = "bitbox-wallet-app";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    hash = "sha256-kvkfz9zwleZNB+leefx+cciJCPkFSSRDxBNRKNqeODc=";
+  };
+  version = "4.40.0";
+  web = buildNpmPackage {
+    pname = "bitbox-web";
+    inherit version;
+    inherit src;
+    sourceRoot = "source/frontends/web";
+    npmDepsHash = "sha256-bnMmeSX8UZpndHK0NMLQhX1GSunO0JSyZdUTfG+rSpY=";
+    installPhase = "cp -r build $out";
+  };
+in
+stdenv.mkDerivation {
+  pname = "bitbox";
+  inherit version;
+
+  inherit src;
+
+  patches = [
+    ./desktop.patch
+    ./genassets.patch
+  ];
+
+  dontConfigure = true;
+
+  buildPhase = ''
+    ln -s ${web} frontends/web/build
+    export GOCACHE=$TMPDIR/go-cache
+    make -C frontends/qt/server linux
+    make -C frontends/qt base
```

Comment:
734649a0811e86d4a1beaf1b0f4702a830f0559d.

I cannot use the toplevel make because i'm using different builders for different parts of the app, namely `buildNpmPackage` for the web part and `mkDerivation` for the Go library and the Qt frontend.